process/block/preprocess/transactions.go

```diff
@@ -874,9 +877,8 @@ func (txs *transactions) createAndProcessMiniBlocksFromMe(
 			}
 		}
 
-		txMaxTotalCost := txs.getTxMaxTotalCost(tx)
-
 		if txs.balanceComputation.IsAddressSet(tx.GetSndAddr()) {
+			txMaxTotalCost = txs.getTxMaxTotalCost(tx)
```

Comment:
I know that it works like this but the code might easily break in panic if someone decides to change the condition on L880 & L973. Strongly suggest to extract `txs.balanceComputation.IsAddressSet(tx.GetSndAddr())` in a bool var that will be used on the both `if` lines

Reply:
Refactored

Reply:
great